### PR TITLE
Add github ci workflow

### PR DESCRIPTION
**Pull request by Lobarr**
- [x] I have read [Chapter's contributing guidelines](https://github.com/freeCodeCamp/chapter/blob/master/CONTRIBUTING.md).
- [x] My pull request has a descriptive title (not a vague title like `Update README.md`).
- [x] My pull request targets the `master` branch of Chapter.

Closes #XXXXX

Description of changes
- Integrated github action for CI

### CODE REVIEW

File: .github/workflows/ci-workflow.yaml [hunk 1]
```diff
@@ -0,0 +1,23 @@
+name: CI Workflow 
+on: [push, pull_request]
+jobs:
+  build:
+    name: build and test
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
```
Comment (ScottBrenner) [inline .github/workflows/ci-workflow.yaml]: No need to use a `matrix` strategy if there's only one version of Node being tested against.

File: .github/workflows/ci-workflow.yaml [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+        node-version: [11.x]
+    steps:
+      - uses: actions/checkout@v1
+      - name: Use Node.js ${{ matrix.node-version }}
+        uses: actions/setup-node@v1
+        with: 
+          node-version: ${{ matrix.node-version }}
+      - name: Install dependencies
+        run: npm ci
```
Comment (ScottBrenner) [inline .github/workflows/ci-workflow.yaml]: I don't know much about JavaScript stuff, but I imagine the commands used here should match  what's in https://github.com/freeCodeCamp/chapter/blob/93f6aad27c81a86f655f27ca63d9069a442d06ff/package.json#L9-L23

File: .github/workflows/ci-workflow.yaml [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+      - name: Build next.js
+        run: npm run next:build
+      - name: Run tests
+        run: npm run test
```
Comment (ScottBrenner) [inline .github/workflows/ci-workflow.yaml]:
```suggestion
        NODE_ENV=test npx jest --coverage --verbose
```

based on https://github.com/freeCodeCamp/chapter/blob/93f6aad27c81a86f655f27ca63d9069a442d06ff/package.json#L21

File: .github/workflows/ci-workflow.yaml [hunk 4]
```diff
@@ -0,0 +1,23 @@
+name: CI Workflow 
+on: [push, pull_request]
+jobs:
+  build:
+    name: build and test
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        node-version: [11.x]
```
Comment (ScottBrenner) [inline .github/workflows/ci-workflow.yaml]:
```suggestion
        node-version: [10.x]
```

based on https://github.com/freeCodeCamp/chapter/blob/bea3dcc1c8174daa3dc942b59380e1cf8c67b373/docker-compose.yml#L4

Recommend reworking the below steps in run in a `node:10-alpine` Docker container as well.

File: .github/workflows/ci-workflow.yaml (hunk 1 above)
Comment (Lobarr) [reply .github/workflows/ci-workflow.yaml]: Included this incase testing multiple versions would be required later. Could remove if not.

**Comment by ScottBrenner** [comment]
The workflow doesn't actually run, since the repository is configured so that
> Only repository collaborators may create workflows.

so there's no way to see if it actually works 😬

File: .github/workflows/ci-workflow.yaml (hunk 3 above)
Comment (Lobarr) [reply .github/workflows/ci-workflow.yaml]: The `npm run test` script, in this case, would be equivalent to the one defined in the package.json hence would resolve to `NODE_ENV=test npx jest --coverage --verbose`

File: .github/workflows/ci-workflow.yaml (hunk 2 above)
Comment (vaibhavsingh97) [reply .github/workflows/ci-workflow.yaml]: @ScottBrenner `npm ci` install a project with a clean slate. For more info: https://docs.npmjs.com/cli/ci.html.

File: .github/workflows/ci-workflow.yaml (hunk 3 above)
Comment (vaibhavsingh97) [reply .github/workflows/ci-workflow.yaml]: Agreed with @Lobarr 👍

Comment (ScottBrenner) [reply .github/workflows/ci-workflow.yaml]: Cool, got it!

File: .github/workflows/ci-workflow.yaml [hunk 5]
```diff
@@ -0,0 +1,23 @@
+name: CI Workflow 
+on: [push, pull_request]
+jobs:
+  build:
+    name: build and test
+    runs-on: ubuntu-latest
+    strategy:
```
Comment (ScottBrenner) [inline .github/workflows/ci-workflow.yaml]:
```suggestion
    strategy:
      fail-fast: false
```

Am a fan of disabling the [`fail-fast`](https://help.github.com/en/github/automating-your-workflow-with-github-actions/workflow-syntax-for-github-actions#jobsjob_idstrategyfail-fast) strategy, to let the full workflow run even when something fails early. Suppose it's up for debate.

**Comment by timmyichen** [comment]
Does github CI support docker?  Later when we have integration tests with a test database we'd need to be able to run our full docker setup to properly test it.

File: .github/workflows/ci-workflow.yaml (hunk 5 above)
Comment (vaibhavsingh97) [reply .github/workflows/ci-workflow.yaml]: what's the use case here? suppose we have test cases are running and if any of the test cases fails then CI should fail, correct me if I am wrong here

**Comment by ScottBrenner** [comment]
Yes, see https://help.github.com/en/github/automating-your-workflow-with-github-actions/workflow-syntax-for-github-actions#jobsjob_idservices.

**Comment by timmyichen** [comment]
Oh, neato - thanks!

File: .github/workflows/ci-workflow.yaml (hunk 5 above)
Comment (Lobarr) [reply .github/workflows/ci-workflow.yaml]: I'm okay with this too so I'd add it.

Comment (vaibhavsingh97) [reply .github/workflows/ci-workflow.yaml]: @Lobarr It's open to discussion, you don't necessarily have to add it

Comment (ScottBrenner) [reply .github/workflows/ci-workflow.yaml]: This is helpful with the `matrix` strategy, say the version matrix is like
```
        node-version: [9.x, 10.x, 11.x]
```

When `fail-fast` is `true`, say anything fails in the `9.x` builds - it cancels all the other builds. But we're still interested in the results for other versions, no? Disabling `fail-fast` continues all in-progress jobs if any `matrix` job fails.

Comment (vaibhavsingh97) [reply .github/workflows/ci-workflow.yaml]: Ohh! Gotcha - Thanks 👍

**Comment by nik-john** [comment]
@all-contributors please add @Lobarr   for platform

**Comment by nik-john** [comment]
@all-contributors please add @Lobarr for platform

**Pull request merged**